mesh_renderer/rasterize_triangles.py

```diff
@@ -103,27 +103,35 @@ def rasterize_clip_space(clip_space_vertices, attributes, triangles,
     raise ValueError('Image height must be > 0.')
   if len(clip_space_vertices.shape) != 3:
     raise ValueError('The vertex buffer must be 3D.')
-  batch_size = clip_space_vertices.shape[0].value
+
   vertex_count = clip_space_vertices.shape[1].value
 
-  per_image_barycentric_coordinates = []
-  per_image_vertex_ids = []
-  for im in range(clip_space_vertices.shape[0]):
+  batch_size = tf.shape(clip_space_vertices)[0]
+  
+  per_image_barycentric_coordinates = tf.TensorArray(dtype=tf.float32, size=batch_size)
+  per_image_vertex_ids = tf.TensorArray(dtype=tf.int32, size=batch_size)
+
+  def condition(b, *args):
```

Comment:
maybe name this function something more descriptive, like "batch_loop_condition"